adam_core/utils/multiprocessing.py

```diff
@@ -0,0 +1,36 @@
+import multiprocessing as mp
+
+
+def _check_parallel(num_jobs):
+    """
+    Helper function to determine how many workers (jobs) should be submitted.
+    If num_jobs is not "auto" and instead an integer then
+    that number will be used instead.
+
+    Parameters
+    ----------
+    num_jobs : {"auto", int}
+        Number of jobs to launch.
+
+    Returns
+    -------
+    enable_parallel : bool
+        True if parallelization should be used (true for cases where num_jobs > 1).
+    num_workers : int
+        The number of workers to use.
```

Comment:
`enable_parallel` doesn't provide a lot more over returning just `num_workers`. Callers could check if `num_workers > 1`. I'd recommend trimming that down, I think it will feel simpler.

Also I'm not convinced that this adds much value in general, yeah. Callers could just use `mp.cpu_count()` themselves, this API is a bit indirect.

Reply:
Removed, turns out the `mp.Pool` accepts both an int and None. If its None then it will use `mp.cpu_count()` processes.